src/app/parcels/ActionBar/ActionModals/ShoppingListModal.tsx

```diff
@@ -110,11 +80,25 @@ const ShoppingListModal: React.FC<ActionModalProps> = (props) => {
         void sendAuditLog({
             action: "create shopping list pdf",
             wasSuccess: false,
-            content: { parcelIds: props.selectedParcels.map((parcel) => parcel.parcelId) },
+            content: { parcelIds: parcelIds },
             logId: pdfError.logId,
         });
     };
 
+    useEffect(() => {
+        const fetchData = async (): Promise<void> => {
+            const { postcodes, error } = await getParcelPostcodesByEvent(
+                "Shopping List Downloaded",
+                parcelIds
+            );
+            if (error) {
+                return;
+            }
+            setDuplicateDownloadedPostcodes(postcodes);
+        };
+        fetchData();
+    }, [parcelIds]);
```

Comment:
The issue I had was that you can't `await` anything in a client component, which I would need to do for this helper function to return the `string[]` I want instead of a Promise. So I tried to get around this by using an effect and state. Do you know of a better workaround?

Reply:
Ah, I see. You can still factor out the "guts" of the function to a helper, and then wrap the call in a function:
```
    const fetchData = async (): Promise<void> => {
        setDuplicateDownloadedPostcodes(
            await getDuplicateDownloadedPostcodes(parcelIds, "Shipping Labels Downloaded")
        );
    };

    fetchData();
```
(no need for the useEffect, i think)

It might feel a little silly but it gives better transparency where `getDuplicateDownloadedPostcodes` is used if someone changes it

Reply:
Or even 
```

    let duplicateDownloadedPostcodes = null;

    const fetchData = async (): Promise<void> => {
        duplicateDownloadedPostcodes = await getDuplicateDownloadedPostcodes(
            parcelIds,
            "Shipping Labels Downloaded"
        );
    };

    fetchData();
```
and remove the state

Reply:
Ok I think I've managed to do it! It turned out the effect (& state) were required so that it updates when you change the selected parcels, otherwise it just fetches once when the page loads (so doesn't work!). But I've made it as simple as possible.

Reply:
Makes sense, looks great now :)